sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -66,6 +66,34 @@ public int getSize() {
         return headers.size();
     }
 
+    /**
+     * Adds a {@link HttpHeader header} with the given name and value if a header with that name doesn't already exist,
+     * otherwise adds the {@code value} to the existing header.
+     *
+     * @param name The name of the header.
+     * @param value The value of the header.
+     * @return The updated HttpHeaders object.
+     */
+    public HttpHeaders add(String name, String value) {
+        if (name == null) {
+            return this;
+        }
+        String caseInsensitiveName = formatKey(name);
+        if (value == null) {
+            return this;
+        } else {
```

Comment:
Does this mean you cannot "reset" the value of a header that's already there? Is there a `remove()` method?

Reply:
Yes, there is a remove method: https://github.com/Azure/azure-sdk-for-java/pull/23888/files/129fffc730b6ceadce1d77d7f80b5016ef798140#diff-21b21d909779e219c34ba1995c680ce1781b4db6bfec0a728055e8532fe36aa1L160